modules/oe_content_featured_media_field/src/Plugin/Field/FieldWidget/FeaturedMediaEntityBrowserWidget.php

```diff
@@ -0,0 +1,269 @@
+        $field_name_key = count($triggering_element['#parents']) - $field_name_key;
+        // Since we are using deltas as well, we go one level below.
+        $field_name_key--;
+        $is_relevant_submit &= ($triggering_element['#parents'][$field_name_key] === $this->fieldDefinition->getName()) &&
+          (array_slice($triggering_element['#parents'], 0, count($element['#field_parents'])) == $element['#field_parents']);
+      }
+    };
+
+    if ($is_relevant_submit) {
+      // Submit was triggered by hidden "target_id" element when entities were
+      // added via entity browser.
+      if (!empty($triggering_element['#ajax']['event']) && $triggering_element['#ajax']['event'] == 'entity_browser_value_updated') {
+        $parents = $triggering_element['#parents'];
+      }
+      // Submit was triggered by one of the "Remove" buttons. We need to walk
+      // few levels up to read value of "target_id" element.
+      elseif ($triggering_element['#type'] == 'submit' && strpos($triggering_element['#name'], $this->fieldDefinition->getName() . '_remove_') === 0) {
+        $parents = array_merge(array_slice($triggering_element['#parents'], 0, -static::$deleteDepth), ['target_id']);
+      }
+
+      if (isset($parents) && $value = $form_state->getValue($parents)) {
+        return EntityBrowserElement::processEntityIds($value);
+      }
+
+      return $entities;
+    }
+
+    // Determine if we are adding a new delta value.
+    if ($triggering_element) {
+      $last_parent = end($triggering_element['#parents']);
+      if ($last_parent === 'add_more') {
+        $parents = $triggering_element['#parents'];
+        // Remove the button key.
+        array_pop($parents);
+        $value = $form_state->getValue($parents);
+        return EntityBrowserElement::processEntityIds($value[$delta]['target_id']);
+      }
+    };
+
+    // We are loading for for the first time so we need to load any existing
+    // values that might already exist on the entity. Also, remove any leftover
+    // data from removed entity references.
+    if (isset($items[$delta]->target_id)) {
+      $entity = $entity_storage->load($items[$delta]->target_id);
+      if (!empty($entity)) {
+        $entities[] = $entity;
+      }
+    }
+    return $entities;
+  }
+
+  /**
+   * Overrides \Drupal\Core\Field\WidgetBase::formMultipleElements().
+   *
+   * Do not allow access to the weight select in order to avoid mixed data
+   * between media items and captions.
+   */
+  protected function formMultipleElements(FieldItemListInterface $items, array &$form, FormStateInterface $form_state) {
+    $elements = parent::formMultipleElements($items, $form, $form_state);
+
+    $field_name = $this->fieldDefinition->getName();
+    $cardinality = $this->fieldDefinition->getFieldStorageDefinition()->getCardinality();
+    $parents = $form['#parents'];
+
+    // Determine the number of possible items.
+    switch ($cardinality) {
```

Comment:
Why you complicating this so much?

1. There is a #max_delta value in the $elements array that tells you the max delta
2. Anyway, it doesn't matter. You can just loop through Element::children() of elements and for each that has a "_weight", we cut the access no?